components/Donate.tsx

```diff
@@ -1,30 +1,35 @@
-"use client";
+'use client';
+import { useCallback, useEffect, useState } from 'react';
+import { parseUnits } from 'viem';
+import { Button } from '@/components/ui/button';
+import { Input } from '@/components/ui/input';
+import { useGrid } from './GridProvider';
```

Comment:
This GridProvider is confusing, I would call it `upProvider` or `clientProvider`